packages/client/src/utils/authUtils.ts

```diff
@@ -90,10 +92,15 @@ export async function refreshToken() {
         token
       })
     })
-    const data = await res.json()
 
-    removeToken()
-    storeToken(data.token)
+    if (!res.ok) {
+      const { store } = createStore()
+      store.dispatch(redirectToAuthentication())
+    } else {
+      const data = await res.json()
+      removeToken()
+      storeToken(data.token)
+    }
```

Comment:
Creating a new store just for redirecting the user to login feels like a code smell. One way to improve this by pushing the error handling to where this function is called from (inside ProtectedPage.tsx). In that component you have direct access to the application store and could dispatch `redirectToAuthentication` there